pkg/controller/podgc/gc_controller_test.go

```diff
@@ -135,90 +150,203 @@ func TestGCTerminated(t *testing.T) {
 
 		gcc.gc()
 
-		pass := true
-		for _, pod := range deletedPodNames {
-			if !test.deletedPodNames.Has(pod) {
-				pass = false
-			}
-		}
-		if len(deletedPodNames) != len(test.deletedPodNames) {
-			pass = false
-		}
-		if !pass {
-			t.Errorf("[%v]pod's deleted expected and actual did not match.\n\texpected: %v\n\tactual: %v", i, test.deletedPodNames, deletedPodNames)
+		if pass := compareStringSetToList(test.deletedPodNames, deletedPodNames); !pass {
+			t.Errorf("[%v]pod's deleted expected and actual did not match.\n\texpected: %v\n\tactual: %v",
+				i, test.deletedPodNames.List(), deletedPodNames)
 		}
 	}
 }
 
 func TestGCOrphaned(t *testing.T) {
-	type nameToPhase struct {
-		name  string
-		phase v1.PodPhase
+	type podMeta struct {
+		name     string
+		nodeName string
+		phase    v1.PodPhase
+	}
+	type nodeEvent struct {
```

Comment:
I guess, same here. Why can't we have explicit fields:
```
nodesToAdd []*v1.Node
nodesToDelete []*v1.Node
```

Reply:
We could, but the beauty of that is deceptive.
nodesToAddToClient, nodesToDeleteFromClient, nodesToAddToInformer, nodesToDeleteFromInformer?

Reply:
You kind of already has it - you're just hiding this in the "event.Type". being more explicit and avoiding switches below sounds like a win to me.
Also, you're not handling e.g. update event type, so that will at least make it explicit.

Reply:
OK